app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomPresenter.kt

```diff
@@ -265,19 +271,37 @@ class ChatRoomPresenter @Inject constructor(
         launchUI(strategy) {
             view.showLoading()
             try {
-                val fileName = async { uriInteractor.getFileName(uri) }.await() ?: uri.toString()
-                val mimeType = async { uriInteractor.getMimeType(uri) }.await()
-                val fileSize = async { uriInteractor.getFileSize(uri) }.await()
-                val maxFileSize = settings.uploadMaxFileSize()
-
-                when {
-                    fileName.isNullOrEmpty() -> view.showInvalidFileMessage()
-                    fileSize > maxFileSize -> view.showInvalidFileSize(fileSize, maxFileSize)
-                    else -> {
-                        Timber.d("Uploading to $roomId: $fileName - $mimeType")
-                        retryIO("uploadFile($roomId, $fileName, $mimeType") {
-                            client.uploadFile(roomId, fileName!!, mimeType, msg, description = fileName) {
-                                uriInteractor.getInputStream(uri)
+                withContext(DefaultDispatcher) {
+                    val fileName = uriInteractor.getFileName(uri) ?: uri.toString()
+                    val fileSize = uriInteractor.getFileSize(uri)
+                    val mimeType = uriInteractor.getMimeType(uri)
+                    val maxFileSizeAllowed = settings.uploadMaxFileSize()
+
+                    when {
+                        fileName.isEmpty() -> {
```

Comment:
Any reason to not use the single line indentation here?

Reply:
Yes. 

It is allowed.
It is on our code base as well.
It isn't an "avoid recommendation".
Like `if(something) in the same line` or `if(something) {in a new line}`

Reply:
Actually it is a "recommendation", but it was more about curiosity 😛

>Put short branches on the same line as the condition, without braces.
```kotlin
when (foo) {
    true -> bar() // good
    false -> { baz() } // bad
}
```